src/racer/core.rs

```diff
@@ -412,21 +413,25 @@ impl IndexedSource {
     }
 
     fn cache_lineoffsets(&self) {
-        if self.lines.borrow().len() == 0 {
-            let mut result = Vec::new();
-            let mut point = 0;
-            for line in self.code.split('\n') {
-                result.push((point, line.len() + 1));
-                point += line.len() + 1;
+        if self.lines.read().unwrap().len() == 0 {
+            let mut lines = self.lines.write().unwrap();
```

Comment:
Is the assumption with all of the `unwrap` calls that poisoning will not occur? It seems like poisoning is a real risk in case of a panic while holding the lock, but there's not much risk of corruption even in such a case.

Should we instead handle poisoning with `.unwrap_or_else(|e| e.into_inner())`?